Makefile

```diff
@@ -58,25 +64,31 @@ pshell: .installed
 # Run development server
 .PHONY: run
 run: .installed
-	@poetry run pserve etc/development.ini
+	@poetry run pserve etc/development.ini --reload
+
+# Testing and linting targets
+all = false
 
 # Testing and linting targets
 .PHONY: lint
-lint: .installed
-	@poetry run pre-commit run --all-files --hook-stage push
+lint: .installed types
+# 1. get all unstaged modified files
+# 2. get all staged modified files
+# 3. get all untracked files
+# 4. run pre-commit checks on them
```

Comment:
A short form could be `git diff --name-only origin/master`

Also, practically It's much difficult to run lint on all the files in CI/CD because one flake8 update can make the whole project incompatible. In those cases we may end up ignoring the added rules in the update for the whole project instead of fixing them. It's much easier to incrementally check for linting issue even in CI/CD with the command `git diff --name-only origin/master`.

Reply:
Also, we can add `poetry check` tests.

Reply:
OK, I'll do it one by one:

1. `git diff --name-only origin/master`: not good enough, because when I run `make lint` I want it to run on all changed files in my workingdir, even those that are not tracked yet.
2. `In those cases we may end up ignoring the added rules in the update for the whole project`: If I ever had a strong opinion against, it's this time! This sweeps the warnings under the rug and we think all is good when in fact it isn't. Exactly the reason why I want to run linters on *all* files in CI is that people don't do this "sweeping under the rug"! If a flake8 update is too invasive, pin the old version in the lockfile, so you know there's something left than needs cleanup.
3. `poetry check` good catch, will add!

Reply:
1. I'm sure `git diff --name-only origin/master` lists also the untracked files.
2. `This sweeps the warnings under the rug and we think all is good when in fact it isn't`: I think we can afford this. Because linting only catches best practices and cleanup issues which are best done incrementally than all at once. `If a flake8 update is too invasive, pin the old version in the lockfile, so you know there's something left than needs cleanup.`: What happened with ebn is there's now a big pending action item to unlock flake8 and the only way to do this is again: fixing everything incrementally.

Reply:
1. My terminal says otherwise, and I just tried again.
2. Yes, incremental fixes are better, buy you have to control the steps. With pinning versions and excluding certain checks, you know exactly how far from "all clear" you are. With randomly silencing checks in certain files and not knowing where and when, you know nothing. 

Additionally, with random silencing, it can happen that you edit one line in one file, and suddenly get 10 warnings for lines you did not even touch. Which is super frustrating and makes you just ignore the whole linter and as such introduce one more warning to the existing warnings (the one in your line).

Reply:
1. My bad... Confused "untracked files" with "untracked changes"
2. The problem is the bigger the project, the more difficult it gets chasing the "all clear" point. And before we reach it, flake8 just releases another update and it's frustrating. I guess I'd rather fix all the files I touched in every pr than owning a user story to fix all the linting issues at once. Although I'd be happy to own that if it's auto fixable like black 😎

Reply:
The big problem with EBN is that it fell behind state-of-the-art and now it's hard to catch up. With woocart, kai, pyramid_openapi3 and pyramid-realworld-example-app we were more diligent (experienced?) and they don't have problems at all with keeping up.